java/core/lance-jni/src/error.rs

```diff
@@ -47,6 +47,10 @@ pub enum Error {
     Arrow { message: String, location: Location },
     #[snafu(display("Index error: {}, location", message))]
     Index { message: String, location: Location },
+    #[snafu(display("Dataset not found error: {}", path))]
+    DatasetNotFound { path: String },
```

Comment:
can we also add location for these two ? so that we can know where the error was raised.